File: server_play_akka_scala/app/domain/models/LunchOffer.scala
package domain.models

import org.joda.money.Money
import java.time.LocalDate
import scala.math.Ordered.orderingToOrdered
import util.PlayDateTimeHelper._

case class LunchOffer(
  id: Id,
  name: String,
  day: LocalDate,
  price: Money,
  provider: LunchProviderId) extends Ordered[LunchOffer] {

  def compare(that: LunchOffer): Int = (this.provider, this.day) compare (that.provider, that.day)
}

File: src/main/kotlin/lunchbox/domain/models/LunchOffer.kt
package lunchbox.domain.models

import java.time.LocalDate
import org.joda.money.Money

typealias LunchOfferId = Long

/**
 * Mittagsangebot
 */
data class LunchOffer(
  val id: LunchOfferId,
  val name: String,
  val details: String,
  val day: LocalDate,
  val price: Money,
  val tags: Set<String>,
  val provider: LunchProviderId
)
